### PR TITLE
Intermezzo DifferentialEquations.jl and Optim

### PR DESCRIPTION
**Pull request by Beramos**
I implemented a small example of a ball bouncing ball to showcase DifferentialEquations.jl and did an experimental fit with Optim.

**Includes:**
* Solving a second order DiffEq
* Using callback functions (event handling) a really nice feature of DifferentialEquations.jl
* Calibrating a parameter using Optim
* Using the VSC code IDE because the animations are pretty

 
image

I plan on adding a few slides showing what DiffEq covers and also what Optim.jl covers

### CODE REVIEW

File: notebooks/day2/intermezzo-diffeq-optim.jl [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  return sol.t , hcat(sol.u...)[1,:]
+end
+
+time, exp_disp = compute_ODE((9.81, 5e-2))
+
+scatter(time, exp_disp, label="Experiment", xlabel="Time(s)", ylabel="Displacement (m)")
+
+function objective(Cd) 
+  _, sim_disp = compute_ODE((9.81, Cd[1]))
+  return sum((sim_disp.- exp_disp).^2)
```
Comment (MichielStock) [inline notebooks/day2/intermezzo-diffeq-optim.jl:68]: I recently learned that `sum(abs2, sim_disp.- exp_disp)` works as a tidy alternative

**Comment by Beramos** [comment]
Ah yes, I sometimes forget but still might be better to keep the original
for clarity in the DS. We should not overflow their brains with tricks.

Op wo 20 jan. 2021 09:29 schreef Michiel Stock <notifications@github.com>:

File: notebooks/day2/intermezzo-diffeq-optim.jl [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+### Optim.jl
+using Optim
+
+function compute_ODE(p; times=1:0.5:15)
+  prob = ODEProblem(f₂, u₀,tspan, p)
+  sol = solve(prob, Tsit5(), saveat=times, callback = cb)
+  return sol.t , hcat(sol.u...)[1,:]
+end
+
+time, exp_disp = compute_ODE((9.81, 5e-2))
```
Comment (MichielStock) [inline notebooks/day2/intermezzo-diffeq-optim.jl:62]: maybe add some noise?
`exp_disp .+= 0.5randn(length(exp_disp))`

**Review by MichielStock — APPROVED** [review]

**Comment by Beramos** [comment]
Good idea!

Op wo 20 jan. 2021 09:38 schreef Michiel Stock <notifications@github.com>:

**Pull request merged**